services/src/datasets/external/nfdi/mod.rs

```diff
@@ -0,0 +1,1031 @@
+    > {
+        let (_, md) = self.dataset_info(dataset).await.map_err(|e| {
+            geoengine_operators::error::Error::DatasetMetaData {
+                source: Box::new(e),
+            }
+        })?;
+
+        let object = self.get_single_file_object(dataset).await.map_err(|e| {
+            geoengine_operators::error::Error::DatasetMetaData {
+                source: Box::new(e),
+            }
+        })?;
+
+        match md.data_type {
+            DataType::SingleVectorFile(info) => {
+                let result_descriptor = Self::create_vector_result_descriptor(md.crs.into(), &info);
+                let template = Self::vector_loading_template(info.layer_name, &result_descriptor);
+
+                let res = NFDIMetaData {
+                    object_id: object.id,
+                    template,
+                    result_descriptor,
+                    _phantom: Default::default(),
+                    object_stub: self.object_stub.clone(),
+                };
+                Ok(Box::new(res))
+            }
+            DataType::SingleRasterFile(_) => Err(geoengine_operators::error::Error::InvalidType {
+                found: md.data_type.to_string(),
+                expected: "SingleVectorFile".to_string(),
+            }),
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl MetaDataProvider<GdalLoadingInfo, RasterResultDescriptor, RasterQueryRectangle>
+    for NFDIDataProvider
+{
+    async fn meta_data(
+        &self,
+        dataset: &DatasetId,
+    ) -> geoengine_operators::util::Result<
+        Box<dyn MetaData<GdalLoadingInfo, RasterResultDescriptor, RasterQueryRectangle>>,
+    > {
+        let (_, md) = self.dataset_info(dataset).await.map_err(|e| {
+            geoengine_operators::error::Error::DatasetMetaData {
+                source: Box::new(e),
+            }
+        })?;
+
+        let object = self.get_single_file_object(dataset).await.map_err(|e| {
+            geoengine_operators::error::Error::DatasetMetaData {
+                source: Box::new(e),
+            }
+        })?;
+
+        match &md.data_type {
+            DataType::SingleRasterFile(info) => {
+                let result_descriptor = Self::create_raster_result_descriptor(md.crs.into(), info);
+                let template = Self::raster_loading_template(info, &result_descriptor);
+
+                let res = NFDIMetaData {
+                    object_id: object.id,
+                    template,
+                    result_descriptor,
+                    _phantom: Default::default(),
+                    object_stub: self.object_stub.clone(),
+                };
+                Ok(Box::new(res))
+            }
+            DataType::SingleVectorFile(_) => Err(geoengine_operators::error::Error::InvalidType {
+                found: md.data_type.to_string(),
+                expected: "SingleRasterFile".to_string(),
+            }),
+        }
```

Comment:
For single files, the 15-minute-key should be fine. We have to find another approach for multiple files and large (chunked) ogr sources.

Reply:
I think we need to discuss this with Marius